### ISSUE TITLE
[Docs] Document Best 2vs3 Practices for Packages providing Python Extensions

### ISSUE THREAD

**Issue by citibeth** [LLNL/spack#645]
See @tgamblin at https://github.com/LLNL/spack/issues/556

> @citibeth: is there a reason you can't get basically what you want, right now, by using variants? Suppose you do this:
> 
> ```
> A:
>     variant('python', default=True, 'enable python bindings')
>     depends_on('python@2.7', when='+python')
>     def patch(self):
>         # do whatever is necessary to enable/disable python bindings according to variant
> B:
>     variant('python', default=True, 'enable python bindings')
>     depends_on('python@3.2:', when='+python')
>     def patch(self):
>         # do whatever is necessary to enable/disable python bindings according to variant
> C:
>     depends_on('A~python')
>     depends_on('B~python')
> ```
> 
> That will result in potentially more copies of A and B than you want but it will work.

**Comment by citibeth** [comment]
Am I correct that dependency types make this issue obsolete?  So, the following should work?

```
A:
    depends_on('python@2.7, when='+python', type='run')
B:
    depends_on('python@3:, when='+python', type='run')

C:
    depends_on('A')
    depends_on('B')
```

**Comment by citibeth** [comment]
Section added in #1676

**Comment by adamjstewart** [comment]
It sounds like this issue was resolved by #1676. Reopen if I'm wrong.
